nucypher/blockchain/eth/sol/source/contracts/PolicyManager.sol

```diff
@@ -146,11 +165,52 @@ contract PolicyManager is Upgradeable {
     }
 
     /**
-    * @notice Set the minimum reward that the node will take
+    * @notice Set range for the minimum reward rate for all nodes
+    */
+    // TODO # 1501
+    // function setMinRewardRateRange(Range calldata _range) external onlyOwner {
+    function setMinRewardRateRange(uint256 _min, uint256 _default, uint256 _max) external onlyOwner {
+        require(_min <= _default && _default <= _max);
+        minRewardRateRange = Range(_min, _default, _max);
+        emit MinRewardRateRangeSet(msg.sender, _min, _default, _max);
+    }
+
+    /**
+    * @notice Set the minimum reward acceptable by node
+    * @dev Input value must be within `minRewardRateRange`
     */
     function setMinRewardRate(uint256 _minRewardRate) external {
-        NodeInfo storage node = nodes[msg.sender];
-        node.minRewardRate = _minRewardRate;
+        require(_minRewardRate >= minRewardRateRange.min &&
+            _minRewardRate <= minRewardRateRange.max,
+            "The min reward rate must be within permitted range");
+        NodeInfo storage nodeInfo = nodes[msg.sender];
+        if (nodeInfo.minRewardRate == _minRewardRate) {
+            return;
+        }
+        nodeInfo.minRewardRate = _minRewardRate;
+        emit MinRewardRateSet(msg.sender, _minRewardRate);
+    }
+
+    /**
+    * @notice Get the minimum reward rate acceptable by node
+    */
+    function getMinRewardRate(NodeInfo storage _nodeInfo) internal returns (uint256) {
+        // if minRewardRate has not been set or is outside the acceptable range
+        if (_nodeInfo.minRewardRate == 0 ||
+            _nodeInfo.minRewardRate < minRewardRateRange.min ||
+            _nodeInfo.minRewardRate > minRewardRateRange.max) {
+            return minRewardRateRange.defaultValue;
```

Comment:
So if a node proactively chooses a rate outside the range, their rate is instead reported to the network as `defaultValue`? That's not a problem, just want to make sure the documentation I write is correct.

Reply:
Not really, node can't set values outside the range, but if we change range and some old min values will be outside then default will be used + initial case when staker didn't set value